src/server/list_family.cc

```diff
@@ -854,6 +854,14 @@ OpResult<string> BPopPusher::RunSingle(ConnectionContext* cntx, time_point tp) {
       std::array<string_view, 4> arr = {pop_key_, push_key_, DirToSv(popdir_), DirToSv(pushdir_)};
       RecordJournal(op_args, "LMOVE", arr, 1);
     }
+
+    if (shard->blocking_controller()) {
```

Comment:
should not you do it only when the operation succeeds (ie. op_res evaluates to true) ?

Reply:
and if the answer is yes, please add the test case with expired timeout as well.